x-pack/plugins/ingest_manager/server/services/epm/registry/index.ts

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { Response } from 'node-fetch';
+import { URL } from 'url';
+import {
+  AssetParts,
+  AssetsGroupedByServiceByType,
+  CategoryId,
+  CategorySummaryList,
+  KibanaAssetType,
+  RegistryPackage,
+  RegistrySearchResults,
+} from '../../../../common/types';
+import { epmConfigStore } from '../config';
+import { cacheGet, cacheSet } from './cache';
+import { ArchiveEntry, untarBuffer } from './extract';
+import { fetchUrl, getResponse, getResponseStream } from './requests';
+import { streamToBuffer } from './streams';
+
+export { ArchiveEntry } from './extract';
+
+export interface SearchParams {
+  category?: CategoryId;
+}
+
+export const pkgToPkgKey = ({ name, version }: RegistryPackage) => `${name}-${version}`;
+
+export async function fetchList(params?: SearchParams): Promise<RegistrySearchResults> {
+  const { registryUrl } = epmConfigStore.getConfig();
```

Comment:
and access `registryUrl` like
```suggestion
  const { registryUrl } = configService.getConfig().epm;
```